include/zephyr/dt-bindings/dma/stm32_dma.h

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright (c) 2023 STMicroelectronics
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#ifndef ZEPHYR_INCLUDE_DT_BINDINGS_DMA_STM32_H_
+#define ZEPHYR_INCLUDE_DT_BINDINGS_DMA_STM32_H_
+
+/**
+ * @name custom DMA flags for channel configuration
+ * @{
+ */
+/** DMA  cyclic mode config */
+#define DMA_STM32_MODE_NORMAL			(0U << 5)
+#define DMA_STM32_MODE_CYCLIC			(1U << 5)
+
+/** DMA  transfer direction config */
+#define DMA_STM32_MEM_TO_MEM			(0U << 6)
+#define DMA_STM32_MEM_TO_PERIPH			(1U << 6)
+#define DMA_STM32_PERIPH_TO_MEM			(2U << 6)
+#define DMA_STM32_PERIPH_TO_PERIPH		(3U << 6)
+
+/** DMA  Peripheral increment Address config */
+#define DMA_STM32_PERIPH_NO_INC			(0U << 9)
+#define DMA_STM32_PERIPH_INC			(1U << 9)
+
+/** DMA  Memory increment Address config */
+#define DMA_STM32_MEM_NO_INC			(0U << 10)
+#define DMA_STM32_MEM_INC			(1U << 10)
+
+/** DMA  Peripheral data size config */
+#define DMA_STM32_PERIPH_8BITS			(0U << 11)
```

Comment:
Please namespace using VND_DMA, same as https://github.com/zephyrproject-rtos/zephyr/blob/main/include/zephyr/dt-bindings/dma/gd32_dma.h (and possibly align names, as they'll likely be the same). This is also done in a few other places, like vendor specific GPIO flags.